plugin/keyring_vault/vault_keys_container.h

```diff
@@ -0,0 +1,27 @@
+#ifndef MYSQL_VAULT_KEYS_CONTAINER_H
+#define MYSQL_VAULT_KEYS_CONTAINER_H
+
+#include "keys_container.h"
+#include "i_vault_io.h"
+
+namespace keyring
+{
+
+class Vault_keys_container : public Keys_container
+{
+public:
+  Vault_keys_container(ILogger* logger)
+    : Keys_container(logger)
+  {}
+
+  my_bool init(IKeyring_io* keyring_io, std::string keyring_storage_url);
+  virtual IKey* fetch_key(IKey *key);
+
+protected:
```

Comment:
I am having problem with private inheritance in keyring framework. The same argument could be used for keyring::Key - when I developed it - there were no class inheriting from it. However, if I used only private instead of protected it would be really hard for me to inherit from Key class in Vault_key class.